endpoints/openrtb2/auction.go

```diff
@@ -1872,3 +1880,23 @@ func checkIfAppRequest(request []byte) (bool, error) {
 	}
 	return false, nil
 }
+
+func (deps *endpointDeps) setIntegrationType(req *openrtb_ext.RequestWrapper, account *config.Account) error {
+	reqExt, err := req.GetRequestExt()
+	if err != nil {
+		return err
+	}
+	reqPrebid := reqExt.GetPrebid()
+
+	if account == nil {
+		return nil
+	}
+	if reqPrebid == nil {
```

Comment:
You need the `account.DefaultIntegration != ""` check as part of this if condition too as it would be pointless to apply the default integration if it is empty.

The other option would be to change the if condition on L1891 to be `if account == nil || account.DefaultIntegration == ""`. This will allow you to return earlier if default integration isn't provided and then you can get rid of the `account.DefaultIntegration != ""` checks in the subsequent if conditions.